pkg/cmd/issue/lock/lock_test.go

```diff
@@ -41,9 +41,36 @@ func Test_NewCmdLock(t *testing.T) {
 			name: "no flags",
 			args: "451",
 			want: LockOptions{
-				SelectorArg: "451",
+				IssueNumber: 451,
 			},
 		},
+		{
+			name: "issue number argument",
+			args: "451 --repo owner/repo",
+			want: LockOptions{
+				IssueNumber: 451,
+			},
+		},
+		{
+			name: "argument is hash prefixed number",
+			// Escaping is required here to avoid what I think is shellex treating it as a comment.
+			args: "\\#451 --repo owner/repo",
+			want: LockOptions{
+				IssueNumber: 451,
+			},
+		},
+		{
+			name: "argument is a URL",
+			args: "https://github.com/cli/cli/issues/451",
+			want: LockOptions{
+				IssueNumber: 451,
+			},
+		},
+		{
+			name:    "argument cannot be parsed to an issue",
+			args:    "unparseable",
+			wantErr: "invalid issue format: \"unparseable\"",
+		},
```

Comment:
Why not re-using `TestArgParsing`?

Reply:
Probably can, I think I didn't because `unlock` requires `parentCmd` to be set: https://github.com/cli/cli/pull/10811/files/55d3b1eaa4e27acad346707ab91c9eb0e4a32a71#diff-c341e36bd18bb73313231c0bda8d126954f03bd766117351db33a1dba406a185R467.

Happy to revisit `lock` and keep `unlock` separate.